### ISSUE TITLE
[Basic Tutorial] From the Frunk: A Guide to EVTO Terminology

### ISSUE THREAD

**Issue by EVGrokker** [aesculus/EVTO-App-Feedback#254]
[DAG From the Trunk series](http://www.digitalautoguides.com/a-guide-to-ev-terminology/)

---

<h3>From the Frunk: A User's Guide to EVTO Terminology</h3>
<dl>
<dt>Jargon, Lingo</dt>
<dd>The specialized vocabulary of a particular field or social group, especially when viewed as unfamiliar. Speech or writing having unusual or pretentious vocabulary, convoluted phrasing, and vague meaning.<br/>
<i>"Arrgh! I'm just not hip to all the EV jargon!"</i></dd>
</dl>
<p>OK, we admit it. EVTO borrows vocabulary from the world of electric vehicles and GPS apps, stirred together with some app-specific mumbo-jumbo we made up. This handy guide will have you talking like the hip kids on TMC pronto.</p>

<hr/>

<h4>Trips, Segments, Routes and Waypoints</h4>

<dl>

<dt>Automatic charging stop</dt>
<dd>[automatic pin icon] When EVTO automatically calculates routes on your behalf, it adds <i>automatic charging stops</i>. Optimizing a route at a later time may select different automatic charging stops. <br/>
You might like an automatic charging stop so much that you want to make sure it's part of your route. (Hey, who are we to judge your Starbucks habit?) To tell EVTO that you'd like to include an automatic charging stop as a required stop, touch the pin to open the <strong>Waypoint Inspector</strong>, then touch the pin icon, converting it from an automatic charging stop to a manual charging stop.</dd>

<dt>Destination pin</dt>
<dd>[destination pin icon] A <i>destination pin</i> is a special waypoint representing a segment's ending point. Destination pins shown with a black checkerboard pattern represent your trip's final destination. Destination pins shown with a gray checkerboard pattern represent intermediate segment destinations.</dd>

<dt>Downstream</dt>
<dd>A <i>downstream</i> waypoint occurs later in time than a subsequent waypoint farther along the same route. For example, a route's destination is downstream from its origin. <br/> 
<i>"Ahoy, matey, is that your paddle I see floating downstream?"</i> <br/>
See also <i>Upstream</i>.
</dd>

<dt>First World problems</dt>
<dd>Any issues you have with EVTO or Tesla ownership.
</dd>

<dt>Hinge pin</dt>
<dd>[intermediate destination pin icon] A hinge pin is a special waypoint shared by two adjoining segments, acting as a <i>destination pin</i> in the upstream (earlier) segment, then an <i>origin pin</i> in the downstream (following) segment. For a multi-segment road trip, a hinge pin would typically represent an overnight stop, possibly with a destination charger, hopefully in perfect working order, ideally without some Ice Age neanderthal already parked there.</dd>

<dt>ICE</dt>
<dd>Internal Combustion Engine. Frequently used dismissively and perjoratively:<br/>
<i>"The last supercharger spot was ICEd by some fossil fuel-farting numbskull loading up on Slim Jims at the 7-11! Grrr!"</i>
</dd>

<dt>Inflating a Route</dt>
<dd>Advanced EVTO users (OK, geeks) use a technique called <i>Route Inflation</i> to plan single-day excursions starting and ending at the same location. To inflate a route, create a new trip with the same origin and destination. Then, add waypoints for each of the stops along your planned route. Great for scenic drives, delivering holiday fruitcakes, or cruising the area Tesla stores looking for new swag. Not to be confused with <i>Tire Inflation</i> or <i>EV Ego Inflation</i>. 
</dd>

<dt>Leg</dt>
<dd>A route fragment between two waypoints. <br/>
<i>"Check out the legs on that route!"</i>
</dd>

<dt>Origin waypoint</dt>
<dd>[origin pin icon]An <i>origin pin</i> is a special waypoint representing a segment's starting point. EVTO has an informed opinion about your preferred initial SoC (State of Charge), derived from <strong>Settings</strong> dialog. An individual segment's <strong>Origin SoC</strong> may be tweaked in the <strong>Edit Segment Details</strong> dialog.</dd>

<dt>POI</dt>
<dd>[waypoint pin icon] A Point of Interest. <a href="https://en.wikipedia.org/wiki/Biggest_ball_of_twine#Largest_ball_of_sisal_twine_built_by_a_community">The World's Largest Ball of Twine</a>, <a href="https://en.wikipedia.org/wiki/Salem_Sue">World's Largest Holstein Cow</a>, <a href="https://en.wikipedia.org/wiki/Hat_%27n%27_Boots">World's Largest Hat 'n Boots</a>… you get the picture. Click <a href="https://en.wikipedia.org/wiki/List_of_largest_roadside_attractions#United_States_of_America">here</a> if, like us, you can't get enough of this stuff. We're always open to a good excuse to take a road trip.</dd>

<dt>Required waypoint</dt>
<dd>[waypoint pin icon] When EVTO calculates routes, it adds <i>automatic charging stops</i>. To convert an automatic waypoint to a required stop, touch the automatic charging stop pin to open the Waypoint Inspector, then touch the pin icon in the Waypoint Inspector, converting it to a regular charging stop.</dd>

<dt>Route</dt>
<dd>The roads selected by EVTO to get you from point A, your <i>origin</i>, to point B, your <i>destination</i>. EVTO evaluates current road and weather conditions, factoring in vehicle characteristics and your <i>SoC</i> (State of Charge) preferences when calculating an optimum route.</dd>

<dt>Segment</dt>
<dd>A route from an <i>origin</i> to a <i>destination</i>, with optional <i>waypoints</i>. A segment may be part of a larger multi-segment trip. The actual route calculated by EVTO for a segment is dynamic, based on your preferences, your EV's characteristics, and current road and weather conditions.</dd>

<dt>Top-Down Trip Design, Bottom-Up Resource Planning</dt>
<dd>The quickest way to wrestle an extended road trip into shape is to start at the top, with an initial origin and final destination. Add daily stopping points, creating segments. Finally, optimize each segment's resource requirements by letting EVTO assign automatic charger stops.
</dd>

<dt>Trip</dt>
<dd><i>Trips</i> are a collection of one or more <i>segments</i> created with EVTO. The standard and Pro versions of EVTO allow planning a single trip at a time. By purchasing an annual Multi-Trip subscription, multiple trips may be saved on your device.<br/>

<table style="width:100%">
  <tr>
    <th></th>
    <th>Standard</th>
    <th>Pro IAP</th> 
    <th>Multi-Trip Subscription IAP</th>
  </tr>
  <tr>
    <td>Create single-segment trips</td>
    <td>✓</td>
    <td>✓</td> 
    <td>✓</td>
  </tr>
  <tr>
    <td>Create multi-segment trips</td>
    <td></td>
    <td>✓</td> 
    <td></td>
  </tr>
  <tr>
    <td>Current Trips</td>
    <td>One</td>
    <td>One</td> 
    <td>Many</td>
  </tr>
</table>

A <i>single-segment trip</i> is a route calculated by EVTO, starting from point A, its <i>origin</i>, to point B, its <i>destination</i>. Available in all versions of EVTO.

A <i>two-segment round trip</i> has its first segment from point A to point B. The second segment begins with point B as its origin, returning to point A as its destination. Requires Pro version IAP.<br/>

<i>Multi-segment trips</i> are a series of two or more segments connected by <i>hinge pins</i>, with today's destination pin becoming tomorrow's origin pin. When planning an extended road trip, use multi-segment trips to partition your trip into manageable one-day segments. Requires Pro version IAP.</dd>

<dt>Upstream</dt>
<dd>Navigation terminology often borrows colorful nautical phrases harkening back to the glory days of sailing ships on the high seas. <i>Upstream</i> comes to us from a time of backwoods types in canoes on creeks. An upstream waypoint is one occurring earlier along the same route. For example, a segment's origin is upstream of its destination. <br/>
See also <i>Downstream</i>.
</dd>

<dt>Waypoint</dt>
<dd>[waypoint pin icon] A marked point along a route associated with a <i>segment</i>, representing <i>origins</i>, <i>destinations</i>, <i>POIs</i>, or <i>charging stops</i>. A <i>hinge pin</i> is a special waypoint serving as both a destination and origin, linking two segments together as part of a larger road trip.</dd>

</dl>

<hr/>
<h4>Batteries, Chargers and Energy</h4>
<dl>

<dt>Destination charging</dt>
<dd>
Tesla partners with hotels, restaurants, shopping centers and resorts to make charging when you arrive at your destination as simple as charging at home. Pull up and plug into a Tesla wall connector to charge while you shop, dine, or even during an overnight stay.<br/>
<a href="https://www.tesla.com/destination-charging">Click for a Tesla-maintained list of destination charging locations</a>.
</dd>

<dt>Level 1 charging</dt>
<dd>Plugging your EV into an ordinary 120v household outlet. For a Tesla, this means about 4.5 miles of added range per hour of charging. While convenient, this is <i>not</i> why we bought Teslas.</dd>

<dt>Level 2 charging</dt>
<dd>A compromise between zzz-inducing <i>Level 1</i> charging and zippy <i>Level 3</i> charging. <i>Level 2</i> chargers provide your car with 240v. Level 2 allows for a range of charging speeds from 30 to 70 miles of added range per hour, depending on the available amperage. Good for overnight charging at home or destination charging. Just remember to plug Mom's electric dryer back in when you're done.</dd>

<dt>Level 3 charging, aka DC Fast Charging</dt>
<dd>The good stuff when you're on the move. Tesla superchargers can provide from 100 to 300 miles of added range per hour of charging. Zzzapp. Cue the smug Tesla grin.</dd>

<dt>SoC</dt>
<dd>[EVGrokker battery icon] State of Charge. The available energy in your battery pack, usually expressed as a percentage, or more generally, "not enough". A constant source of range anxiety.<br/>
"This SoCs! Why didn't I buy a bigger battery?"
</dd>

![img_0349](https://cloud.githubusercontent.com/assets/26678403/25141062/9066c7ec-241f-11e7-96f9-c8f34e3cfd92.PNG)

<dt>SoL</dt>
<dd>State of [energy] Lusting. The degree of your range anxiety, typically inversely proportional to your battery size.  <br/>
"SoL! Why didn't I buy a bigger battery?"
</dd>

![empty battery](https://cloud.githubusercontent.com/assets/26678403/25140834/c7fddf0c-241e-11e7-9803-0b4363b55548.png)

<dt>SoT</dt>
<dd>State of Tranquility. A rarely-achieved peace of mind for range-anxious EV owners.<br/>
"For a fleeting moment, I felt that I had purchased a large enough battery. But that feeling passed quickly." 
</dd>

![full battery](https://cloud.githubusercontent.com/assets/26678403/25191793/f6dea174-24ed-11e7-8a24-e64040d3be97.png)

</dl>

<hr/>

<p><i>Did we blank your favorite buzzword? Got an idea for a tutorial? <a href="mailto:support@digitalautoguides.com?Subject=Tutorial suggestion">Let us know!</a>
</p>

**Comment by EVGrokker** [comment]
One last thing on the Terminology page:

You need to add the tag at the bottom. Here's the HTML:
```
<hr/>
<p><i>Did we blank your favorite buzzword? Got an idea for a tutorial? <a href="mailto:support@digitalautoguides.com?Subject=Tutorial suggestion">Let us know!</a>
</p>
```

**Comment by aesculus** [comment]
Done

**Comment by EVGrokker** [comment]
To be added to the Trips, Segments, Routes and Waypoints Section:

<img width="102" alt="5 stars" src="https://cloud.githubusercontent.com/assets/26678403/25556181/791588be-2cb5-11e7-85a3-b6939cb422c8.png">

<h4>App Store Ratings</h4>
<p>[5 star icon above]A developer's lifeblood, serving as a means to help new users to your favorite apps. EVTO is one of your favorite apps, right? If we are, tell the world – <a href="https://itunes.apple.com/us/app/ev-trip-optimizer-for-tesla/id1124549080?mt=8">leave us a positive rating</a>. If not, <a href="mailto:support@digitalautoguides.com?Subject=Suggestion box">let us know</a> how we can improve. Seriously.</p>

**Comment by aesculus** [comment]
Should this be at the bottom in a single column just before the line?

**Comment by EVGrokker** [comment]
Alpha order, first!

**Comment by EVGrokker** [comment]
Wherever you want it. Maybe too pushy at the top.

**Comment by aesculus** [comment]
Duh. Now I get it. **App Store Ratings** is a defined term. I thought you were just begging. :-)

Do you think this is the right place for this? It does seem a bit pushy on the terms page.  What about on the main page in the EVTO section or the EVTO Tutorials page?

Rather than a term, just a general plea?

**Comment by EVGrokker** [comment]
I plan on pushing ratings everywhere, see the newest tutorial for an example. 

Compromise? How about naming it 'Ratings (App Store)' so it falls later in the list? I would like it on this page as well as others.

**Comment by aesculus** [comment]
OK. What about your Android friends (OK you don't have any :-)

**Comment by EVGrokker** [comment]
You're right, I have no friends. And if I did, they wouldn't have Androids. Actually, that's not fair, my son has an an Android, but that's only because I won't hand him down my old iPhones when I upgrade. I'm cheap that way.

But seriously, does the Android store use ratings? Does it matter in that world? If so, I would reword the last two sentences:

>If we are, tell the world – leave us a positive rating on the App Store[link] or Google Play[link]. If not, let us know how we can improve. Seriously.

**Comment by EVGrokker** [comment]
Oops. Just noticed a typo in Downstream:

A downstream waypoint occurs later in time than a ~~subsequent~~ previous waypoint, farther along the same route. For example, a route’s destination is downstream from its origin.
“Ahoy, matey, is that your paddle I see floating downstream?”
See also Upstream.

**Comment by aesculus** [comment]
Yes. They do

EVGrokker wrote:
>
> But seriously, does the Android store use ratings? Does it matter in 
> that world? If so, I would reword the last two sentences:
>
>     If we are, tell the world – leave us a positive rating on the App
>     Store[link] or Google Play[link]. If not, let us know how we can
>     improve. Seriously.
>

-- 
Chris Couper

**Comment by EVGrokker** [comment]
RE: Ratings… if you add the Google Play link, then it doesn't make sense to paranthesize the title. In that case, I'd just title it 'Ratings' and leave it at that. I was trying to avoid future ambiguity with Trip Ratings, but that's a perfectly fine title for that future entry that won't collide with Ratings from the respective app stores.

Question: What's the ratio of iOS users to Android users?

**Comment by aesculus** [comment]
Rate the App?

**Comment by EVGrokker** [comment]
If we use 'Rate the App', I'd rewrite the definition as follows:

<img width="102" alt="5 stars" src="https://cloud.githubusercontent.com/assets/26678403/25556181/791588be-2cb5-11e7-85a3-b6939cb422c8.png">

<h4>Rate the App</h4>
<p>[5 star icon above]Successful apps depend on users taking a moment to post ratings. Ratings help us to help you. If you find EVTO useful, please leave us a positive rating <a href="https://itunes.apple.com/us/app/ev-trip-optimizer-for-tesla/id1124549080?mt=8">on the App Store</a> or Google Play [link]. If we're not meeting your expectations, <a href="mailto:support@digitalautoguides.com?Subject=Suggestion box">let us know</a> how we can improve. Seriously.</p>

**Comment by EVGrokker** [comment]
After mulling this over, I think 'Ratings' and the initial definition fit best with the others.

**Comment by aesculus** [comment]
OK. Just so I don't go stupid. Copy and paste your final revision below and let me know exactly where it should go.

**Comment by EVGrokker** [comment]
To be added in **Trips, Segments, Routes and Waypoints**, between _POI_ and _Required Waypoint_:

<img width="102" alt="5 stars" src="https://cloud.githubusercontent.com/assets/26678403/25556181/791588be-2cb5-11e7-85a3-b6939cb422c8.png">

<h4>Ratings</h4>
<p>[use the 5 star icon above]A developer's lifeblood, serving as a means to help new users discover apps others find useful. EVTO is one of your favorite apps, right? If we are, tell the world – leave us a positive rating at <a href="https://itunes.apple.com/us/app/ev-trip-optimizer-for-tesla/id1124549080?mt=8">the App Store</a> or on Google Play [insert link to EVTO on Google Play]. If not, <a href="mailto:support@digitalautoguides.com?Subject=Suggestion box">let us know</a> how we can improve. Seriously.</p>

---
To be updated in **Trips, Segments, Routes and Waypoints**:

<h4>Downstream</h4>
<p>A downstream waypoint occurs later in time than a previous waypoint on the same route. For example, a route’s destination is downstream from its origin.<br/>
<i>“Ahoy, matey, is that your paddle I see floating downstream?”</i><br/>
See also Upstream.</p>

**Comment by aesculus** [comment]
I can only use a single icon image otherwise the format goes off so I just used a standard star.

**Comment by EVGrokker** [comment]
To be added to **Batteries, Chargers and Energy**:

<h4>[circled checkmark] Reachability</h4>
<p>Color coding applied to charging and destination pins, predicting Arrival SoC.</p> 
<table style="width:100%">
  <tr>
    <th>Arrival SoC</th>
    <th>&lt; 5%</th>
    <th>5% to Reserve SoC</th>
    <th>&gt; Reserve SoC</th>
  </tr>
  <tr align="center">
    <td align="right"><strong>Pin Color</strong></td>
    <td>Red</td>
    <td>Yellow</td>
    <td>Green</td>
  </tr>
</table>
<p>The default value for Reserve SoC is defined in Settings. A trip segment's Reserve SoC may be edited in the Edit Segment Details dialog.</p>

**Comment by aesculus** [comment]
Applied

**Comment by EVGrokker** [comment]
To be added to **Trips, Segments, Routes and Waypoints**:
<h4>[scales icon] Optimization</h4>
<p>When a trip is created using Automatic waypoint charger selection, EVTO selects an optimal route based upon supercharger proximity. If you subsequently add a waypoint that causes the route to be changed and recalculated, you should also use the <strong>Optimize Routing</strong> menu item to allow EVTO to reassess the optimal route incorporating your new waypoint.</p> 

<h4>[circled checkmark icon] Recalculation</h4>
<p>When a trip is initially created, EVTO identifies the optimal route based on time and distance. If you specify <i>Automatic</i> waypoint charger selection, it also factors in supercharger locations to route you to your destination. If you subsequently adjust any elements of your itinerary – departure times, Minimum Stop Times at waypoints, adding new waypoints – EVTO will recalculate your itinerary. This affects the arrival and departure times of individual waypoints, but does not affect the trip routing.</p>

**Comment by EVGrokker** [comment]
For the Reachability definition, suggest the following edits to help table layout:
~~Arrival~~ Soc
~~Pin~~ Color

Also, consider reducing font size slightly help headers live on one line in narrow browser windows.

<img width="459" alt="screen shot 2017-05-02 at 9 04 52 am" src="https://cloud.githubusercontent.com/assets/26678403/25624390/6fa8083e-2f16-11e7-8bfb-f4b479f3a72b.png">

**Comment by EVGrokker** [comment]
Edit for SoL entry definition:
State of [energy] Lusting. The degree of your range anxiety, typically inversely proportional to your ~~battery size~~ SoC.

**Comment by aesculus** [comment]
> Also, consider reducing font size slightly help headers live on one line in narrow browser windows.

Would this be for all the headers in the web tutorials or are you talking about the App help headers?

**Comment by EVGrokker** [comment]
>>Also, consider reducing font size slightly help headers live on one line in narrow browser windows.

>Would this be for all the headers in the web tutorials or are you talking about the App help headers?

This comment was specifically for the table headers in the Reachability to help them fit in a reasonably-sized browser window without wrapping.

**Comment by aesculus** [comment]
I changed the text but the font reduction will take some more effort so I will table that for now.

**Comment by aesculus** [comment]
I also updated the charger images with the new style

**Comment by EVGrokker** [comment]
~~**Required Waypoint**~~: Change to **Manual Charging Stop**, then move it to be in the proper order alphabetically. This makes the terminology consistent with the definition for Automatic Charging Stop.

**Comment by aesculus** [comment]
I made this change. The text still says required if you want to review it.

**Comment by EVGrokker** [comment]
Good catch. Please make the following revision to the text:

When EVTO calculates routes, it adds automatic charging stops. 
To convert an automatic waypoint to a required stop, touch the pin to open the Waypoint Inspector, then touch the pin icon in the Waypoint Inspector, converting it to a manual charging stop.

**Comment by aesculus** [comment]
Done
